PoobChat/PoobChat/Login/UserRepository.swift

```diff
@@ -11,18 +11,25 @@ import CoreData
 class UserRepository {
     let context = PersistenceController.shared.container.viewContext
     
-    func saveUser(id: Int64, username: String, creationDate: Date) throws {
+    func saveUser(id: String, username: String, creationDate: Date) throws {
         let user = getUser(id: id) ?? User(context: context)
         user.username = username
         user.id = id
         user.creationDate = Date()
         try context.save()
     }
     
-    func getUser(id: Int64) -> User? {
+    func getUser(id: String) -> User? {
         let request: NSFetchRequest<User> = User.fetchRequest()
-        request.predicate = NSPredicate(format: "id == %i", id)
+        request.predicate = NSPredicate(format: "id == %@", id)
         let results = try? context.fetch(request)
         return results?.first
     }
+    
+    func saveUserDTO(dto: UserResponseDTO) throws {
+        let user = getUser(id: dto.id) ?? User(context: context)
```

Comment:
Looks like you are only saving the username. We need to save the other properties as well (like we are on lines 15 - 20)